pygfx/utils/transform.py

```diff
@@ -0,0 +1,281 @@
+import numpy as np
+import pylinalg as pla
+from datetime import datetime
+
+from typing import List
+
+OUTDATED = datetime(1900, 1, 1)  # some date in the past
+
+
+class AffineBase:
+    last_modified: datetime
+    matrix: np.ndarray
+    position: np.ndarray
+    rotation: np.ndarray
+    scale: np.ndarray
+
+    def __init__(self) -> None:
+        self._components_age = OUTDATED
+        self._matrix: np.ndarray = None
+        self._position: np.ndarray = None
+        self._rotation: np.ndarray = None
+        self._scale: np.ndarray = None
+
+        self._inverse_age = OUTDATED
+        self._inverse_matrix: np.ndarray = None
+
+    def _update_inverse(self):
+        if self.last_modified < self._inverse_age:
```

Comment:
comparing datetime objects is relatively hard work compared to checking a flag. it's going to be hit thousands of times as well. you must be bumping the `last_modified` attribute in several locations, why won't a `dirty` flag do?

Reply:
Great explanation, thanks. This helps me build an understanding of this (pretty complex) code